autoload/jedi.vim

```diff
@@ -288,16 +288,44 @@ function! jedi#goto_definitions() abort
 endfunction
 
 function! jedi#usages() abort
-    call jedi#remove_usages()
     PythonJedi jedi_vim.usages()
 endfunction
 
-function! jedi#remove_usages() abort
-    for match in getmatches()
-        if stridx(match['group'], 'jediUsage') == 0
-            call matchdelete(match['id'])
-        endif
-    endfor
+" Hide usages in the current window.
+function! jedi#hide_usages_in_win() abort
+    if has_key(w:, '_jedi_usages_matchids')
+        for matchid in w:_jedi_usages_matchids
+            call matchdelete(matchid)
+        endfor
+        unlet w:_jedi_usages_matchids
+    endif
+    " Remove the autocommands that might have triggered this function.
+    augroup jedi_usages
+        autocmd! * <buffer>
+    augroup END
+endfunction
+
+function! jedi#show_usages_in_win() abort
+    PythonJedi jedi_vim.highlight_usages_for_win()
+    if !exists('#jedi_usages#TextChanged')
+        augroup jedi_usages
+        " Unset highlights on any changes to this buffer.
+        autocmd TextChanged <buffer> call jedi#clear_usages()
+        " Hide usages when the buffer is removed from the window, or when
+        " entering insert mode (but keep them for later).
+        autocmd BufWinLeave,InsertEnter <buffer> call jedi#hide_usages_in_win()
+        augroup END
```

Comment:
Can you indent everything in the augroup? I think it's way more readable the way you did it above.